observability/test/run_o11y_tests.py

```diff
@@ -0,0 +1,1174 @@
+                                           ServerStartArgs(port = self.args.port))
+            )
+            gke_env.start_testservice_server(self.server_config, WAIT_SECS_SERVER_START,
+                                             server_start_cmd)
+            client_action_cmd = './%s %s' % (
+                self.get_client_action_cmd(self.args.client_lang),
+                self.get_client_action_args(
+                    self.args.client_lang,
+                    ClientActionArgs(server_host = gke_env.server_pod_ip,
+                                     server_port = self.args.port,
+                                     action = action,
+                                     exporter_interval = WAIT_SECS_EXPORTER_FLUSH)))
+            gke_env.run_client_commands(self.client_config, WAIT_SECS_CLIENT_ACTION, [
+                client_action_cmd,
+            ])
+            gke_env.kill_testservice_server()
+            self.wait_before_querying_o11y_data()
+        except Exception:
+            self.test_runner.set_exit_status(1)
+            logger.error(traceback.format_exc())
+        finally:
+            gke_env.clean_up_gke_resources()
+            self.sponge_log_out.close()
+
+    def setup_and_run_rpc_vm(self,
+                             action: str,
+                             config_location: ConfigLocation = ConfigLocation.FILE,
+                             server_config_into_env_var: Optional[ObservabilityConfig] = None,
+                             client_config_into_env_var: Optional[ObservabilityConfig] = None) -> None:
+        if config_location == ConfigLocation.FILE:
+            server_env = self.get_env_with_config(
+                self.server_config,
+                config_file_name = 'server-config-%s.json' % self.identifier)
+            client_env = self.get_env_with_config(
+                self.client_config,
+                config_file_name = 'client-config-%s.json' % self.identifier)
+        else:
+            server_env = self.get_env_with_config(self.server_config)
+            client_env = self.get_env_with_config(self.client_config)
+        if server_config_into_env_var:
+            server_env[CONFIG_ENV_VAR_NAME] = server_config_into_env_var.toJson()
+        if client_config_into_env_var:
+            client_env[CONFIG_ENV_VAR_NAME] = client_config_into_env_var.toJson()
+        server_proc = self.start_server_in_subprocess(env = server_env)
+        time.sleep(WAIT_SECS_SERVER_START)
+        try:
+            client_proc = self.start_client_in_subprocess(action, env = client_env)
+            client_proc.wait(timeout=WAIT_SECS_CLIENT_ACTION)
+            self.wait_before_querying_o11y_data()
+        except Exception:
+            self.test_runner.set_exit_status(1)
+            logger.error(traceback.format_exc())
+        finally:
+            logger.info('Killing server...')
+            os.kill(server_proc.pid, signal.SIGTERM)
```

Comment:
You should probably check the return code and print it out. Segfaults are possible and we need to know when they happen.

Reply:
You're not also going to send `SIGTERM` to the client?

Reply:
So for now, the client is supposed to exit by itself after executing those actions, and flushing the observability logging/metrics/trace to cloud. Will see if this needs to be improved.